File: cortex-spring-boot-autoconfigure/src/main/java/ai/chatur/cortex/spring/dataset/DatasetService.java
package ai.chatur.cortex.spring.dataset;

import org.apache.jena.query.Dataset;
import org.apache.jena.riot.Lang;
import org.apache.jena.riot.RDFDataMgr;
import org.apache.jena.system.Txn;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.stereotype.Service;

@Service
public class DatasetService {
  @Autowired
  @Qualifier("assertions")
  Dataset assertions;

  public void printAssertions() {
    Txn.executeRead(assertions, () -> RDFDataMgr.write(System.out, assertions, Lang.TRIG));
  }
}

File: src/main/kotlin/ai/chatur/cortex/core/DatasetService.kt
package ai.chatur.cortex.core

import java.io.OutputStream
import org.apache.jena.query.Dataset
import org.apache.jena.riot.Lang
import org.apache.jena.riot.RDFDataMgr
import org.apache.jena.system.Txn

class DatasetService(private val ds: Dataset) {

  fun export(os: OutputStream) {
    Txn.executeRead(ds) { RDFDataMgr.write(os, ds, Lang.TRIG) }
  }
}
